tests/apm/monitoring-serverClosed-001.phpt

```diff
@@ -0,0 +1,50 @@
+--TEST--
+MongoDB\Driver\Monitoring\ServerClosedEvent
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$m = create_test_manager();
+
+class MySubscriber implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    private $serverClosed = false;
+
+    public function serverChanged(MongoDB\Driver\Monitoring\ServerChangedEvent $event) {}
+
+    public function serverClosed(MongoDB\Driver\Monitoring\ServerClosedEvent $event)
+    {
+        if ($this->serverClosed) {
+            return;
+        }
+
+        $this->serverClosed = true;
+        echo "- getHost() returns a string: ", is_string($event->getHost()) ? 'yes' : 'no', "\n";
+        echo "- getPort() returns an integer: ", is_integer($event->getPort()) ? 'yes' : 'no', "\n";
+        echo "- getTopologyId() returns an ObjectId: ", ($event->getTopologyId() instanceof MongoDB\BSON\ObjectId) ? 'yes' : 'no', "\n";
+    }
+
+    public function serverOpening(MongoDB\Driver\Monitoring\ServerOpeningEvent $event) {}
+    
+    public function topologyChanged(MongoDB\Driver\Monitoring\TopologyChangedEvent $event) {}
+
+    public function topologyOpening(MongoDB\Driver\Monitoring\TopologyOpeningEvent $event) {}
+}
+
+$subscriber = new MySubscriber;
+MongoDB\Driver\Monitoring\addSubscriber($subscriber);
+
+$command = new MongoDB\Driver\Command(['ping' => 1]);
+$m->executeCommand(DATABASE_NAME, $command);
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+- getHost() returns a string: yes
+- getPort() returns an integer: yes
+- getTopologyId() returns an ObjectId: yes
+===DONE===
```

Comment:
The `monitoring-serverClosed-001.phpt` test fails on the GitHub CI (https://github.com/mongodb/mongo-php-driver/pull/1251/checks?check_run_id=3365180493) because I think no `ServerClosedEvent` is emitted from the 'ping' command - how should I make sure that we get a server closed event?

Reply:
Did this test ever pass locally for you, or was CI the first time you observed it in action?

According to the [SDAM Monitoring spec](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring-monitoring.rst#events), ServerClosedEvent is:

> Published when the server monitor’s connection is closed and the server is shutdown.

You may need to use the `disableClientPersistence` driver option for Manager (third argument) to ensure the connection get closed when the manager object goes out of scope. Without that, the libmongoc client will outlive the Manager object.

While you're doing this, I'd also suggest using `Manager::addSubscriber()` for this test (ahead of [PHPC-1959](https://jira.mongodb.org/browse/PHPC-1959)). Offhand, I'm not sure if adding a subscriber on a specific Manager will add a reference to it, which might prevent it from being garbage collected (and thus the libmongoc client being destroyed due to `disableClientPersistence`). I don't believe that's the case, as `phongo_apm_copy_manager_for_client` may be the only place that is done (and only for the lifetime of the command monitoring events being dispatched). Using `Manager::addSubscriber()` would allow us to confirm that's indeed the case.

Reply:
Ah ok, the test did pass locally for me but not on the CI.

Reply:
Hmm I added the `disableClientPersistence` option but the test seems to fail for the same reason as before (https://github.com/mongodb/mongo-php-driver/pull/1251/checks?check_run_id=3374366170).